packages/components/src/Button.tsx

```diff
@@ -29,7 +29,7 @@ interface BaseButtonProps {
   icon?: IconDefinition | JSX.Element;
   disabled?: boolean;
   active?: boolean;
-  onClick: React.MouseEventHandler<HTMLButtonElement>;
+  onClick?: React.MouseEventHandler<HTMLButtonElement>;
```

Comment:
Right now you could pass `type='button'` and `onClick=undefined` and TypeScript wouldn't report an error - you'd only get an error at run time with prop types.

You'd need to add another interface that extends this, and then use that with `ButtonWithChildren` and `IconOnlyButton` I think. Eg.

```
interface ButtonButtonProps extends BaseButtonProps {
  type?: 'button';
  onClick: React.MouseEventHandler<HTMLButtonElement>;
}

interface SubmitButtonProps extends BaseButtonProps {
  type: 'reset' | 'submit';
  onClick: undefined;  // I think? Or omit if we still want to allow onClick with submit/reset types
}

interface ButtonWithTypeProps = ButtonButtonProps | SubmitButtonProps;

interface ButtonWithChildren extends ButtonWithTypeProps { ... }
interface IconOnlyButton extends ButtonWithTypeProps { ... }

```